### ISSUE TITLE
Logging option

### ISSUE THREAD

**Issue by sureshvv** [pyusb/pyusb#197]
I want to have an option to log all read/writes to a file when using pyusb. Does this make sense? Anyone has done this?

**Comment by jonasmalacofilho** [comment]
Yes, it makes sense, and I'm guessing we're not alone in implementing this ad-hoc.

Can you provide a patch/PR that implements this?  My only note is that it must have ~0 cost unless debug is enabled.
